backend/app.js

```diff
@@ -36,26 +38,64 @@ async function getId() {
 
 async function getInventory(idList, json) {
```

Comment:
this function looks really great! Can you add some inline documentation so that it is more clear to the user what each part does. Feel free to have a block comment at the beginning explaining why all of this is necessary.